src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -345,7 +348,7 @@ module.exports = function<T, P, I, TI, C, CX>(config : HostConfig<T, P, I, TI, C
       firstEffect = finishedWork.firstEffect;
     }
 
-    prepareForCommit();
+    const commitInfo = prepareForCommit();
```

Comment:
Oh right, I forgot that `componentWillUnmount` is called during the first pass. That explains it then.